pr-dhl-woocommerce/includes/class-pr-dhl-blocks-integration.php

```diff
@@ -0,0 +1,372 @@
+
+
+		wp_enqueue_script(
+			$handle,
+			$script_url,
+			$asset['dependencies'],
+			$asset['version'],
+			true
+		);
+
+		wp_set_script_translations(
+			$handle,
+			'dhl-for-woocommerce',
+			PR_DHL_PLUGIN_DIR_PATH . '/languages'
+		);
+	}
+
+	/**
+	 * Registers AJAX actions.
+	 */
+	private function register_ajax_actions() {
+		add_action( 'wp_ajax_pr_dhl_set_checkout_post_data', array( $this, 'handle_set_checkout_post_data' ) );
+		add_action( 'wp_ajax_nopriv_pr_dhl_set_checkout_post_data', array( $this, 'handle_set_checkout_post_data' ) );
+		add_action( 'wp_ajax_pr_dhl_get_preferred_days', array( $this, 'handle_get_preferred_days' ) );
+		add_action( 'wp_ajax_nopriv_pr_dhl_get_preferred_days', array( $this, 'handle_get_preferred_days' ) );
+	}
+
+
+
+	/**
+	 * Returns an array of script handles to enqueue in the frontend context.
+	 *
+	 * @return string[]
+	 */
+	public function get_script_handles() {
+		return [
+			'pr-dhl-preferred-services-integration',
+			'pr-dhl-preferred-services-frontend',
+			'pr-dhl-parcel-finder-frontend',
+		];
+	}
+
+	/**
+	 * Returns an array of script handles to enqueue in the editor context.
+	 *
+	 * @return string[]
+	 */
+	public function get_editor_script_handles() {
+		return [
+			'pr-dhl-preferred-services-integration',
+			'pr-dhl-preferred-services-editor',
+			'pr-dhl-parcel-finder-editor',
+		];
+	}
+
+	/**
+	 * An array of key, value pairs of data made available to the block on the client side.
+	 *
+	 * @return array
+	 */
+	public function get_script_data() {
+		return [
+			'pr-dhl-active' => true,
+		];
+	}
+
+	/**
+	 * Localizes scripts with necessary data.
+	 */
+	private function localize_scripts() {
+		// Fetch the shipping settings
+		$dhl_settings      = PR_DHL()->get_shipping_dhl_settings();
+
+		// Set conditions for parcel finder options.
+		$packstation_enabled = ! empty( $dhl_settings['dhl_display_packstation'] ) && 'yes' === $dhl_settings['dhl_display_packstation'];
+		$parcelshop_enabled  = ! empty( $dhl_settings['dhl_display_parcelshop'] ) && 'yes' === $dhl_settings['dhl_display_parcelshop'];
+		$post_office_enabled = ! empty( $dhl_settings['dhl_display_post_office'] ) && 'yes' === $dhl_settings['dhl_display_post_office'];
+
+		$localize_data = array(
+			'pluginUrl'           => PR_DHL_PLUGIN_DIR_URL,
+			'dhlSettings'         => $dhl_settings,
+			'packstation_enabled' => $packstation_enabled,
+			'parcelshop_enabled'  => $parcelshop_enabled,
+			'post_office_enabled' => $post_office_enabled,
+			'ajax_url'            => admin_url( 'admin-ajax.php' ),
+			'nonce'               => wp_create_nonce( 'pr_dhl_nonce' ),
+			'parcel_nonce'        => wp_create_nonce( 'dhl_parcelfinder' ),
+			'DHL_ENGLISH_REGISTRATION_LINK'        => DHL_ENGLISH_REGISTRATION_LINK,
+			'DHL_GERMAN_REGISTRATION_LINK'        => DHL_GERMAN_REGISTRATION_LINK,
+			'locale' => get_locale(),
+			'google_maps_enabled' => $dhl_settings['dhl_display_google_maps'],
```

Comment:
I got this error when testing this PR : `PHP Warning:  Undefined array key "dhl_display_google_maps" in /var/www/html/wp-content/plugins/pr-dhl-woocommerce/includes/class-pr-dhl-blocks-integration.php on line 268
`